### ISSUE TITLE
After the recent update I can't get up and running. angular 5

### ISSUE THREAD

**Issue by keli5466** [thisissoon/angular-inviewport#28]
<img width="590" alt="screen shot 2018-10-09 at 10 12 11 am" src="https://user-images.githubusercontent.com/8658144/46686022-d351ce00-cbab-11e8-8270-85e6381f60b0.png">

Getting this error above after the update and following the setup directions. I'm really looking to use the new limit & debounce funcationality. 

Before I was using 2.2 because of a compatibility issue and I set it up like this in the app.module.ts

import { InViewportModule, WindowRef} from '@thisissoon/angular-inviewport';

imports: [
    InViewportModule.forRoot([{ provide: WindowRef, useFactory: () => window }])
]

**Comment by edoparearyee** [comment]
@keli5466 I've looked into this issue and the issue seems to be because we are using a new `providedIn` api for angular 6, which breaks if using angular 5.

As a workaround you can add the following to your providers in your app module

```ts
 providers: [
    { provide: WINDOW, useFactory: () => window }
  ]
```
